app/components/BuyQuoteBox.tsx

```diff
@@ -0,0 +1,384 @@
+
+    // Buy Quote Request Parameters & wrapper function to change parameter state
+    const [buyQuoteParams, setBuyQuoteParams] = useState<BuyQuoteRequest>(emptyBuyQuoteParams);
+    const onChangeBuyQuotesParams = (e: ChangeEvent<HTMLSelectElement | HTMLInputElement>) => {
+        const { name, value } = e.target;
+        setBuyQuoteParams(prevState => {
+            return {
+                ...prevState,
+                [name]: value
+            }
+        });
+    }
+    // Buy Quote Response
+    const [buyQuoteResponse, setBuyQuoteResponse] = useState<BuyQuoteResponse>();
+
+    /* Wrapper around buy options API call under api/buy-options-api
+        - Calls and awaits the API with the current buyOptionsParams state
+        - Sets the buyOptionsResponse state to API response, reset buyQuoteParams
+        - handles error and loading states
+    */
+    const buyOptionsWrapper = useCallback(async () => {
+        if (!buyOptionsParams.country) { // ensure country is selected
+            alert("Please select a country to generate buy options!");
+            return;
+        }
+        if (buyOptionsParams.country + buyOptionsParams.subdivision === prevCountrySubdiv.current) { // prevent re-fetching same data
+            return;
+        }
+
+        try {
+            // set loading state, reset current response
+            setIsOptionsLoading(true);
+            setBuyOptionsResponse(undefined);
+            const response = await generateBuyOptions(buyOptionsParams);
+
+            // set response, reset buy quote parameters, 
+            setBuyOptionsResponse(response?.json);
+            setBuyQuoteParams({
+                ...emptyBuyQuoteParams,
+                country: buyOptionsParams.country,
+            });
+
+            // remove loading state and cache last country + subdivision query
+            setIsOptionsLoading(false);
+            prevCountrySubdiv.current = buyOptionsParams.country + buyOptionsParams.subdivision; // store current query params for future caching
+        } catch (error) {
+            // remove loading state and alert user of error
+            setIsOptionsLoading(false);
+            alert(error);
+        }
+    }, [buyOptionsParams]);
+
+
+    /* Wrapper around buy quote API call under api/buy-quote-api
+        - Calls and awaits the API with the current buyQuoteParams state
+        - Sets the buyQuoteResponse state to API response
+    */
+    const buyQuoteWrapper = useCallback(async () => {
+        if (!buyQuoteParams.purchase_currency || !buyQuoteParams.payment_currency || !buyQuoteParams.payment_method || !buyQuoteParams.payment_amount || !buyQuoteParams.country) {
+            alert("Please fill out all required fields");
+            return;
+        }
+        if (parseInt(buyQuoteParams.payment_amount) < parseInt(payment_amount_limits.min) || 
+            parseInt(buyQuoteParams.payment_amount) > parseInt(payment_amount_limits.max)) {
```

Comment:
nit: long conditionals can be a little harder to read
```suggestion
const paymentAmount = parseInt(buyQuoteParams.payment_amount);
const minPaymentAmount = parseInt(payment_amount_limits.min);
const maxPaymentAmount = parseInt(payment_amount_limits.max);
        if (paymentAmount < minPaymentAmount  || paymentAmount > maxPaymentAmount) {
```